grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -749,6 +749,15 @@ class StockMovementService {
             }
             requisitionItems.each { requisitionItem ->
                 StockMovementItem stockMovementItem = StockMovementItem.createFromRequisitionItem(requisitionItem)
+                InventoryItem inventoryItem = inventoryService.findOrCreateInventoryItem(
```

Comment:
This does not feel safe in a method used to GET data. Can we just findInventoryItem(), and if we cannot find one, we either a) ignore and leave inventory item NULL OR b) blow things up with a validation error if that's more appropriate? I assume an inventory item is not always populated at certain points in the stock movement so maybe (a) is fine. 

I think we need to discuss these validation issues again in a tech huddle. I'm confused by what we're trying to accomplish. It seems the issue is on the React side because there's not enough data available which you're trying to solve by creating new data.